sched/generators.go

```diff
@@ -0,0 +1,51 @@
+package sched
+
+import (
+	"fmt"
+	"github.com/leanovate/gopter"
+	//"github.com/leanovate/gopter/gen"
+)
+
+func GenJob(id string, numTasks int) Job {
+	jobDef := JobDefinition{
+		JobType: "testJob",
+		Tasks:   make(map[string]TaskDefinition),
+	}
+
+	// Generate tasks
+	for i := 0; i < numTasks; i++ {
+		task := GenTask()
+		taskId := fmt.Sprintf("Task%v", i)
+		jobDef.Tasks[taskId] = task
+	}
+
+	job := Job{
+		Id:  id,
+		Def: jobDef,
+	}
+
+	return job
+}
+
+func GenTask() TaskDefinition {
+	return TaskDefinition{
+		Command: Command{
+			Argv: []string{"cmd1", "cmd2"},
+		},
+	}
+}
+
+// Randomly generates an Id that is valid for
+// use as a sagaId or taskId
+func GenJobId() gopter.Gen {
+	return func(genParams *gopter.GenParameters) *gopter.GenResult {
```

Comment:
It's super-annoying that gopter.gen package wants you to work with monads in a language that's so hostile (syntactically, and type-system-wise) to monads... : /